spec/tapioca/compilers/dsl/active_record_typed_store_spec.rb

```diff
@@ -0,0 +1,194 @@
+# typed: false
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe(Tapioca::Compilers::Dsl::ActiveRecordTypedStore) do
+  describe("#initialize") do
+    def constants_from(content)
+      with_content(content) do
+        subject.processable_constants.map(&:to_s).sort
+      end
+    end
+
+    it("gathers no constants if there are no ActiveRecordTypedStore classes") do
+      expect(subject.processable_constants).to(be_empty)
+    end
+
+    it("gather only TypedStore classes") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.string(:reviewer, blank: false, accessor: false)
+          end
+        end
+        class Shop < ActiveRecord::Base
+        end
+
+        class User
+        end
+      RUBY
+
+      expect(constants_from(content)).to(eq(["Post"]))
+    end
+  end
+
+  describe("#decorate") do
+    def rbi_for(content)
+      with_content(content) do
+        parlour = Parlour::RbiGenerator.new(sort_namespaces: true)
+        subject.decorate(parlour.root, Post)
+        parlour.rbi
+      end
+    end
+
+    it("generate RBI for TypedStore classes without accessor") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadatado do |s|
+            s.string(:reviewer, blank: false, accessor: false)
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+
+      RUBY
+
+      expect(rbi_for(content)).to(eq(expected))
+    end
+
+    it("generate RBI for TypedStore classes with accessor") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.string(:reviewer, blank: false, accessor: true)
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Post
+          sig { returns(T.nilable(String)) }
+          def reviewer; end
+
+          sig { params(reviewer: T.nilable(String)).returns(T.nilable(String)) }
+          def reviewer=(reviewer); end
+
+          sig { returns(T::Boolean) }
+          def reviewer?; end
+
+          sig { returns(T.nilable(String)) }
+          def reviewer_before_last_save; end
+
+          sig { returns(T.nilable([T.nilable(String), T.nilable(String)])) }
+          def reviewer_change; end
+
+          sig { returns(T::Boolean) }
+          def reviewer_changed?; end
+
+          sig { returns(T.nilable(String)) }
+          def reviewer_was; end
+
+          sig { returns(T.nilable([T.nilable(String), T.nilable(String)])) }
+          def saved_change_to_reviewer; end
+
+          sig { returns(T::Boolean) }
+          def saved_change_to_reviewer?; end
+        end
+       RUBY
+
+      expect(rbi_for(content)).to(eq(expected))
+    end
+
+    it("generate RBI for TypedStore classes with fields null and default as false") do
```

Comment:
```suggestion
    it("generates methods for accessors marked as non-null with non-nilable types") do
```

Reply:
the title of the test should read as an English sentence and describe which situation is being tested, not what the test data is.

So `it("generate RBI for TypedStore classes with fields null and default as false")` does not conform to either of those.

Something like `it("generates methods with non-nilable types for accessors marked as not null")` describes what is being tested better and reads like a proper English sentence.